### DISCUSSION TITLE
Skip authorization check on Edge Functions

### DISCUSSION THREAD

**Discussion by deliqs** [supabase/supabase#8569] [Questions] [answered]
Hi,

I'd like to connect some webhooks to certain Edge Functions on my project, but the Edge Function return a 401 (Unauthorized). I unfortunately cannot add auth headers to the web hook. Is there any way around this?

**Comment by kevinclarkadstech** [comment, marked as answer]
> Hi,
> 
> 
> 
> I'd like to connect some webhooks to certain Edge Functions on my project, but the Edge Function return a 401 (Unauthorized). I unfortunately cannot add auth headers to the web hook. Is there any way around this?

That would be a big security hole, depending on what the function does. You usually can attach some sort of state query parameter to web hooks and OAuth redirects. Could you do that to at least create your own sort of API key?

**Reply by deliqs** [reply]
It's about validating a completed purchase from Stripe. This has its own validation call in it, but I can understand the security risks in other cases. Adding middleware would fix it, but I'm not looking for that. I'll try to do it via another way. Thanks!

**Reply by betastreets** [reply]
https://supabase.com/docs/guides/functions#deploy-to-production

By default, Edge Functions require a valid JWT in the authorization header. This header is automatically set when invoking your function via a Supabase client library.

If you want to use Edge Functions to handle webhooks (e.g. Stripe payment webhooks etc.), you need to pass the --no-verify-jwt flag when deploying your function.

**Reply by deliqs** [reply]
Thanks for the answer. I found the solution a short while after this question. So I luckily didn't have to create middleware and unnecessary bloat.

**Reply by bpeck81** [reply]
> https://supabase.com/docs/guides/functions#deploy-to-production
> 
> By default, Edge Functions require a valid JWT in the authorization header. This header is automatically set when invoking your function via a Supabase client library.
> 
> If you want to use Edge Functions to handle webhooks (e.g. Stripe payment webhooks etc.), you need to pass the --no-verify-jwt flag when deploying your function.

I literally wanted to make a simple change and forgot about this flag. You saved me hours since it would've been a while before I looked at the docs.

**Reply by justinfrevert** [reply]
There's certainly plenty of use cases for allowing the verification to be turned off. Teams rolling their own auth, read-only data, etc. I notice that when I deploy with the setting mentioned above, or even set it manually, I still receive the error for the JTW check.

**Reply by flogy** [reply]
It would be awesome if we could have some configuration file for the edge functions that can be added to version control. Like that we could disable the JWT check for specific functions and it will never reset it when deploying said function, even if the `--no-verify-jwt` was not added. What do you think?

**Reply by flogy** [reply]
> It would be awesome if we could have some configuration file for the edge functions that can be added to version control. Like that we could disable the JWT check for specific functions and it will never reset it when deploying said function, even if the `--no-verify-jwt` was not added. What do you think?

Just realized this is now possible: https://supabase.com/docs/reference/cli/config#functions.function_name.verify_jwt

E.g. add the following to your `config.toml` to disable verifying JWT token for the function `helloworld`:

```
[functions.helloworld]
verify_jwt = false
```

**Reply by garylake** [reply]
@flogy thanks for this, even better, the config.toml solution works when serving them locally too 👍🏻

**Reply by alijawad1511** [reply]
@deliqs can you please tell me how you resolved the above issue you mentioned? I am stuck at that issue of 401 error

**Reply by justinekatebe** [reply]
> config.toml

goto your config.toml file, and add this code to your file. *Please note that you have to replace "function_name" with the name of the function you want to skip jwt auth.

```
[functions.function_name]
verify_jwt = false
```

**Reply by AebadUQ** [reply]
https://<project_id>.supabase.co/functions/v1/oauth-facebook
api status is pending and not giving me response
  fetch("https://<project_id>.supabase.co/functions/v1/oauth-facebook", {
        method: "POST",
        
        body: JSON.stringify({ code }),
      })
        .then((res) => res.json())
        .then((data) => {
          console.log("Connection success", data);
        //   navigate("/dashboard");
        })
        .catch(console.error);
    }
